docs/content/development/extensions-core/kafka-ingestion.md

```diff
@@ -122,6 +122,7 @@ The tuningConfig is optional and default parameters will be used if no tuningCon
 |`indexSpec`|Object|Tune how data is indexed, see 'IndexSpec' below for more details.|no|
 |`buildV9Directly`|Boolean|Whether to build a v9 index directly instead of first building a v8 index and then converting it to v9 format.|no (default == true)|
 |`reportParseExceptions`|Boolean|If true, exceptions encountered during parsing will be thrown and will halt ingestion; if false, unparseable rows and fields will be skipped.|no (default == false)|
+|`publishTimeout`|Long|Milliseconds to wait for publishing segments. It must be >= 0, where 0 means to wait forever.|no (default == 0)|
```

Comment:
I think there is no need of `publishTimeout` as if the publish is failing it should be retried infinitely. This is because if one publish fails then subsequent publishes will anyways fail. There is already pending completion timeout config as Supervisor which takes care that the task does not keep on running infinitely.